frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -968,6 +970,7 @@ const ManageHostsPage = ({
           : `Hosts successfully transferred to  ${team.name}.`;
 
       dispatch(renderFlash("success", successMessage));
+      setResetSelectedRows(true); // TODO: Consider this an an alternative approach to setClearSelectionCount
```

Comment:
Ah, so this is the **right** way to do it :) Increment the `clearSectionCount` felt a little hacky to me. 

If we are going to remove the calls to `setClearSelectionCount()` can we remove that useState variable, and also remove any instances of that in the props? It's passed through `TableContainer` and `DataTable` so will need to be removed there, as well.

Reply:
I'm not sure this is the right way either. It is based on the `TableContext` approach that is being used in `CoreLayout`. But it depends on toggling the `autoResetSelectedRows` property of the react-table instance (see line 143 of `DataTable`) and that feels a bit too remote for me to trust it will works as expected without unintended side effects in all cases. I liked your idea of the `DataTable` listening for changes to a value in a `useEffect` hook and firing the `toggleAllRowsSelected` methods exposed by react-table rather than toggling the auto reset property. Maybe we could craft an approach that uses `TableContext` as the dependency for the `useEffect` hook?

Reply:
That could work. Since release day is tomorrow my inclination is to leave this change as-is since the code changes are minimal. It seems pretty solid to me, and since we're using `dispatch` and redux states to broadcast lots of global events (like `FlashMessage`) it doesn't strike me as a bad pattern for the time being.

Reply:
@lukeheath We are going to have multiple tables rendered in the same page so context would need to be able to differentiate which table selection state is being changed. So we could look at designing that into context. Or maybe it is better to manage it at the page level as a prop that can be separately managed on each `TableContainer`/`DataTable` instance. I had some trouble intuiting what `clearSelectionCount` was supposed to do and why it was an incrementing number. Maybe we could instead make it `shouldClearSelectedRows, setShouldClearSelectedRows` and we could have the `useEffect` in `DataTable` listen for that boolean to change to `true` and then fire `toggleAllRowsSelected` to `false` and then `setShouldClearSelectedRows` back to `false`.

Reply:
@lukeheath sounds good. I don't think the multiple tables issue is a current issue as I can't think of places where we have select actions in two tables on the same page.  I'll make the clean up changes you suggested and then merge. Assuming this fix passes QA, we can follow up on the alternate approach after the release.

Reply:
Sounds good! And all good points above re: multiple tables, and `shouldClearSelectedRows` not being very clear. If your current implementation works for this release we can go ahead with that, and then have a bit more time to design a proper long-term solution. I'll create an engineering-driven ticket to track the need to rework the DataTable.